api/include/opentelemetry/std/span.h

```diff
@@ -26,6 +26,9 @@
 #      define HAVE_SPAN
 #    endif
 #  endif
+#  if !__has_include(<gsl/gsl>)
```

Comment:
@lalitb #813 - if it's not available, the option is undefined.

Reply:
Thanks @maxgolov. Seems fine. We can discuss this PR changes (i.e making absl::variant as default) in today's community meeting.

Reply:
So, if `gsl:span` is available, and `HAVE_CPP_STDLIB`option is set, for C++11/14/17, we use gsl::span instead of nostd::span. I don't think that is the correct approach. We should fallback to nostd::span unless customer explicitly wants to use any other library?

Reply:
Ok. have gone through #821 - should be fine as long as we are documenting it well.

Reply:
I ti